apartmentsbot/apartment_manager.py

```diff
@@ -0,0 +1,46 @@
+from telebot.states import State, StatesGroup
+from telebot.storage import StateMemoryStorage
+from utility import get_all_apartments
+from parser.apartment import Apartment
+
+state_storage = StateMemoryStorage()
+
+
+class MyStates(StatesGroup):
+    city = State()
+    price = State()
+    area = State()
+    rooms = State()
+    next_data = State()
+
+
+class ApartmentManager:
+    def __init__(self):
+        self.apartments = []
+        self.index = 0
+
+    def load_apartments(self, city: str, area: int, max_price: int, rooms: int) -> None:
+        self.apartments = self.execute_filter(city, area, max_price, rooms)
+        self.index = 0
+
+    def get_next_batch(self, batch_size=5) -> list[Apartment]:
+        start = self.index
+        end = start + batch_size
+        self.index = end
+        return self.apartments[start:end]
```

Comment:
What if end index is higher than `self.apartments` len?

Reply:
Fixed it.